### PR TITLE
Make configuration options more visible in documentation

### LINKED ISSUE #1758: Document need for --oversubscribe with new MPI versions
**Describe the flaw**
This is a follow-up to #1725. With new MPI versions, running `make installcheck` may result in failing MPI tests when the computer running the tests has too few cores. This can be fixed easily, but not generally (see comment by @jougs in #1725) by adding the `--oversubscribe` option.

**Suggest an improvement**
Installation documentation (near `make installcheck`) should point out that this problem may occur and that it can be fixed by adding `--oversubscribe` to the `mpirun` command in `.nestrc`. Documentation for that file and for running NEST with MPI might also want to point this out.

### PR DESCRIPTION
**Pull request by sarakonradi**
This PR fixes #1758. 

More specifically, it:
- Splits the compilation and configuration options into two separate pages
- Adds "Configure" to the table of contents
- Links the compilation options in ``installation/index.rst``
- Mentions the Mailing List Archive and Stack Overflow in ``troubleshooting.rst``

### CODE REVIEW

File: doc/installation/configuration.rst [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+NEST configuration file
+-----------------------
+
+Upon importing the NEST Python module or starting the ``nest`` executable from the command line, NEST will create a
+configuration file called ``.nestrc`` in your home directory.
+
+Using this file, you can set a number of options:
+
+* The browser for displaying the helpdesk
+* The pager for showing the built-in help
```
Comment (jougs) [inline doc/installation/configuration.rst:15]:
```suggestion
* The pager for showing the built-in help in the terminal
```

File: doc/installation/configuration.rst [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+* The ``mpirun`` command for :doc:`parallel execution <../guides/parallel_computing>`
```
Comment (jougs) [inline doc/installation/configuration.rst:16]:
```suggestion
* The ``mpirun`` command for :doc:`parallel execution <../guides/parallel_computing>` of the testsuite
```

File: doc/installation/configuration.rst [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+In case your MPI Implementation requires special options (e.g. ``--oversubscribe`` to allows the use of more
+processes than compute cores in OpenMPI versions above 3.0), you can add them to the ``mpirun`` command as shown in
```
Comment (jougs) [inline doc/installation/configuration.rst:19]:
```suggestion
processes than available compute cores in OpenMPI versions above 3.0), you can add them to the ``mpirun`` command as shown in
```

File: doc/installation/configuration.rst [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+Command line switches for the nest executable
+---------------------------------------------
+
+Type
+
+::
+
+   nest --help
+
+to find out about NEST’s command-line parameters.
```
Comment (jougs) [inline doc/installation/configuration.rst:42]:
```suggestion
to find out about NEST's command-line parameters.
```

File: doc/troubleshooting.rst [hunk 5]
```diff
@@ -3,14 +3,19 @@ Troubleshooting
...
 
+Please make sure you have:
 
-1. CMAKE error says a <package> was not found or <package> is too old
----------------------------------------------------------------------
-
+* Followed the :doc:`installation instructions <installation/index>`
+* Installed the required :ref:`dependencies <standard>`
+* Viewed the :doc:`compilation <installation/compilation_options>` and 
+  :doc:`configuration <installation/configuration>` options
```
Comment (jougs) [inline doc/troubleshooting.rst:11]: In my view the items after the bullet points should all start with a lower case letter and the final one should end in a full stop. But as we seem to use the same form as here throughout the whole documentation, I don't think this PR is the place to change anything but I rather wanted to take the opportunity to voice my opinion. Feel free to ignore it :wink:

**Review by jougs — COMMENTED** [review]
Very nice and sorry for the long delay in reviewing this. See my suggestions below for some ideas for minor improvements.

File: doc/troubleshooting.rst (hunk 5 above)
Comment (sarakonradi) [reply doc/troubleshooting.rst:11]: I agree with you here! However, for the sake of consistency with the remaining documentation, I would keep it as it is for now. We can address the style in a separate issue  :+1:.

File: doc/sli_docs/an-introduction-to-sli.rst [hunk 6]
```diff
@@ -30,84 +26,15 @@ at the command prompt. You should then see something like this:
...
 
-1. `First Steps <first-steps.md>`__
-2. `Objects and data types <objects-and-data-types.md>`__
-3. `Programming in SLI <programming-in-sli.md>`__
-4. `Using files and keyboard
-   input <using-files-and-keyboard-input.md>`__
-5. `Neural simulations <neural-simulations.md>`__
+1. :doc:`First Steps <first-steps>`
+2. :doc:`Objects and data types <objects-and-data-types>`
+3. :doc:`Programming in SLI <programming-in-sli.md>`
```
[the remark this answers is not in the document]
Comment (sarakonradi) [reply doc/sli_docs/an-introduction-to-sli.rst]: Done.

**Comment by sarakonradi** [comment]
@jougs Thanks for the precise review. I have applied your suggestions and responded to your last comment inline.

**Review by jougs — APPROVED** [review]
Thanks for addressing all my comments! There's still a conflict that needs resolving before merging

**Comment by sarakonradi** [comment]
Done!

**Pull request merged**